### PR TITLE
[sortablejs] Fix types for the onMove callback

### PR DESCRIPTION
**Pull request by stof**
While return 1 or -1 is supported to override the final position, returning nothing is also supported (keeping the default position which is based on the dragging direction).
To be exact, any returned value other than `false`, `1` and `-1` is treated the same (keeping the default behavior).

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://github.com/SortableJS/Sortable/blob/88838bfa5647e278f4cf87fdd4c3e34cd24fb33c/src/Sortable.js#L1247-L1254>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## sortablejs ([<kbd>unpkg</kbd>](https://unpkg.com/browse/sortablejs@latest/))
was missing the following properties:
1. version
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 340f67daeee374a4c4860155469a10ca6f6aeb44
</p>

**Comment by peterblazejewicz** [comment]
This is legit code, but the public api tells to return specific values:
https://github.com/SortableJS/Sortable/tree/88838bfa5647e278f4cf87fdd4c3e34cd24fb33c#options  (`onMove`)
https://github.com/SortableJS/Sortable/blob/88838bfa5647e278f4cf87fdd4c3e34cd24fb33c/Sortable.js#L2491
Should we rather tend to follow public api here?

**Comment by stof** [comment]
@peterblazejewicz I opened an issue on the library to ask the maintainers to clarify the intended signature.

**Review by sandersn — CHANGES_REQUESTED** [review]
Couple of thoughts from a DT maintainer/Typescript engineer:

1. This seems like an unintended effect of the implementation Sortable.js uses in _onDragOver. Since this type specifies the callback the user is supposed to provide to sortablejs, I think it's best to strictly represent the documented API instead of representing ANY value that would actually work.
3. `void` should only be used as the single return type of a function -- it's basically in the language for backward compatibility with C-languages. `undefined` is `void`'s supertype, but here, `unknown` is technically correct. _onDragOver has SOME behaviour for any value. However, `unknown | X` collapses to `unknown`, which would hide the useful types `1 | -1 | boolean`.

Given this, I think it's best to (1) not make this change or (2) just add `undefined` to the return type to allow the new test to pass, since `undefined` is falsy.

**Comment by qlonik** [comment]
I'm curious, how did I end up being one of the people in the commit? I have never touched anything to do with `sortablejs`.

**Comment by waynevanson** [comment]
Hi Guys, maintainer here.

@sandersn Has a great point, but I think the docs don't reflect it correctly.

I think when this was implemented, the signature of `false | 1 | -1 | void` was intended.
This should be reflected in the docs.

related https://github.com/SortableJS/Sortable/issues/1861

File: types/sortablejs/index.d.ts
Comment (waynevanson) [inline types/sortablejs/index.d.ts:412]:
```suggestion
        onMove?: (evt: MoveEvent, originalEvent: Event) => false | -1 | 1 | void;
```

**Comment by stof** [comment]
@waynevanson Based on https://github.com/SortableJS/Sortable/issues/1861#issuecomment-662748869, the implementation I provided seems to be the good one.

**Review by waynevanson — APPROVED** [review]
LGTM! Is inline with the implementation in Sortable.

Thank you for your advices, @sandersn :)

**Comment by stof** [comment]
@sandersn FYI, the doc of SortableJS has now been updated. It would be great if you could unblock this PR (I think your review requesting changes is the reason why the bot is not merging)

**Review by sandersn — APPROVED** [review]
I'd still prefer to have `boolean | -1 | 1 | undefined`, but I think `void` is good enough.

**Pull request merged**
